flagd/pkg/service/flag-evaluation/ofrep/handler.go

```diff
@@ -0,0 +1,115 @@
+package ofrep
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/open-feature/flagd/core/pkg/evaluator"
+	"github.com/open-feature/flagd/core/pkg/logger"
+	"github.com/open-feature/flagd/core/pkg/service/ofrep"
+	"github.com/rs/xid"
+)
+
+const (
+	key              = "key"
+	singleEvaluation = "/ofrep/v1/evaluate/flags/{key}"
+	bulkEvaluation   = "/ofrep/v1/evaluate/{path:flags\\/|flags}"
+)
+
+type handler struct {
+	Logger    *logger.Logger
+	evaluator evaluator.IEvaluator
+}
+
+func NewHandler(logger *logger.Logger, evaluator evaluator.IEvaluator) http.Handler {
+	h := handler{
+		logger,
+		evaluator,
+	}
+
+	router := mux.NewRouter()
+	router.HandleFunc(singleEvaluation, h.HandleFlagEvaluation).Methods("POST")
+	router.HandleFunc(bulkEvaluation, h.HandleBulkEvaluation).Methods("POST")
+	return router
+}
+
+func (h *handler) HandleFlagEvaluation(w http.ResponseWriter, r *http.Request) {
+	requestID := xid.New().String()
+
+	// obtain flag key
+	vars := mux.Vars(r)
+	flagKey := vars[key]
+	if vars == nil || strings.TrimSpace(flagKey) == "" {
+		h.writeJSONToResponse(
+			http.StatusInternalServerError,
+			ofrep.InternalError{ErrorDetails: "failed to obtain the flag key from the request"}, w)
+		return
+	}
+
+	request := ofrep.Request{}
+	err := json.NewDecoder(r.Body).Decode(&request)
+	if err != nil && err.Error() != "EOF" {
+		// Special handling for invalid context
+		h.writeJSONToResponse(http.StatusBadRequest, ofrep.ContextErrorResponseFrom(flagKey), w)
+		return
+	}
+
+	context := map[string]any{}
+	if res, ok := request.Context.(map[string]any); ok {
+		context = res
+	} else {
+		h.Logger.WarnWithID(requestID, "provided context does not comply with flagd, continuing ignoring the context")
+	}
+
+	evaluation := h.evaluator.ResolveAsAnyValue(r.Context(), requestID, flagKey, context)
+	if evaluation.Error != nil {
+		status, evaluationError := ofrep.ValueToStatusAndError(evaluation)
+		h.writeJSONToResponse(status, evaluationError, w)
+	} else {
+		h.writeJSONToResponse(http.StatusOK, ofrep.SuccessResponseFrom(evaluation), w)
+	}
+}
+
+func (h *handler) HandleBulkEvaluation(w http.ResponseWriter, r *http.Request) {
+	requestID := xid.New().String()
+
+	request := ofrep.Request{}
+	err := json.NewDecoder(r.Body).Decode(&request)
+	if err != nil && err.Error() != "EOF" {
+		// Special handling for invalid context
+		h.writeJSONToResponse(http.StatusBadRequest, ofrep.ContextErrorResponseForBulkAndStatus(), w)
+		return
+	}
+
+	context := map[string]any{}
+	if res, ok := request.Context.(map[string]any); ok {
+		context = res
+	} else {
+		h.Logger.WarnWithID(requestID, "provided context does not comply with flagd, continuing ignoring the context")
+	}
```

Comment:
this is repeated, maybe we could export it into its own private function 🤔

Reply:
Done, common logic was extracted to helper functions :)